src/components/TodoItem.tsx

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import React, {
+  useContext, useEffect, useState,
+  FormEvent, KeyboardEvent, useRef,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoContext } from '../contexts/TodoContext';
+import { deleteTodo, updateTodo } from '../api/todos';
+import { Error } from '../types/Error';
+
+interface Props {
+  todo: Todo;
+}
+
+type EditEvent = FormEvent<HTMLFormElement>
+| KeyboardEvent<HTMLInputElement> | null;
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    setTodos,
+    setErrorMessage,
+    loadingTodo,
+    setLoadingTodo,
+  } = useContext(TodoContext);
+
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const myInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    myInput.current && myInput.current.focus();
+  }, [isEditing, myInput]);
+
+  const handleEditStart = () => {
+    setIsEditing(true);
+  };
+
+  const handleEditCancel = () => {
+    setIsEditing(false);
+    setNewTitle(title);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodo(todo);
+
+    deleteTodo(todoId)
+      .then(() => setTodos((prev) => prev.filter((t) => t.id !== todoId)))
+      .catch(() => setErrorMessage(Error.Delete))
+      .finally(() => setLoadingTodo(null));
+  };
```

Comment:
you have duplicated code (when delete todo)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/248d4af1-b15b-4242-b7ed-580678dc8325)